modules/home-environment.nix

```diff
@@ -96,52 +96,18 @@ in
   meta.maintainers = [ maintainers.rycee ];
 
   options = {
-    home.file = mkOption {
-      description = "Attribute set of files to link into the user home.";
-      default = {};
-      type = types.loaOf (types.submodule (
-        { name, config, ... }: {
-          options = {
-            target = mkOption {
-              type = types.str;
-              description = ''
-                Path to target file relative to <envar>HOME</envar>.
-              '';
-            };
-
-            text = mkOption {
-              default = null;
-              type = types.nullOr types.lines;
-              description = "Text of the file.";
-            };
-
-            source = mkOption {
-              type = types.path;
-              description = ''
-                Path of the source file. The file name must not start
-                with a period since Nix will not allow such names in
-                the Nix store.
-                </para><para>
-                This may refer to a directory.
-              '';
-            };
-
-            mode = mkOption {
-              type = types.str;
-              default = "444";
-              description = "The permissions to apply to the file.";
-            };
-          };
-
-          config = {
-            target = mkDefault name;
-            source = mkIf (config.text != null) (
-              let name' = "user-etc-" + baseNameOf name;
-              in mkDefault (pkgs.writeText name' config.text)
-            );
-          };
-        })
-      );
+    home.username = mkOption {
+      type = types.str;
+      default = builtins.getEnv "USER";
```

Comment:
Unfortunately I think it is a bad idea to put this in the default field (same with `HOME` below). Mainly because the documentation will show "root" and "/root" if Home Manager is installed as a system package.

Instead could put `home.username = mkDefault (builtins.getEnv "USER");` in the `config` section.